pytorch_lightning/metrics/functional/accuracy.py

```diff
@@ -0,0 +1,117 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Tuple, Optional
+
+import torch
+from pytorch_lightning.metrics.classification.helpers import _input_format_classification
+
+
+def _accuracy_update(
+    preds: torch.Tensor, target: torch.Tensor, threshold: float, top_k: Optional[int], mdmc_accuracy: str
+) -> Tuple[torch.Tensor, torch.Tensor]:
+
+    preds, target, mode = _input_format_classification(preds, target, threshold=threshold, top_k=top_k)
+
+    if mode in ["binary", "multi-label"]:
+        correct = (preds == target).all(dim=1).sum()
+        total = target.shape[0]
+    elif mdmc_accuracy == "global":
+        correct = (preds * target).sum()
+        total = target.sum()
+    elif mdmc_accuracy == "subset":
+        extra_dims = list(range(1, len(preds.shape)))
+        sample_correct = (preds * target).sum(dim=extra_dims)
+        sample_total = target.sum(dim=extra_dims)
+
+        correct = (sample_correct == sample_total).sum()
+        total = target.shape[0]
+
+    return (torch.tensor(correct, device=preds.device), torch.tensor(total, device=preds.device))
```

Comment:
`correct` should always be on the correct device (I think), so no need to do `torch.tensor` on that.
For `total` only two of the cases needs casting (first and last). Please convert in these two cases such that the return statement simply becomes:
```suggestion
    return correct, total
```